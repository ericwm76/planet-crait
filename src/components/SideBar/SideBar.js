import React from 'react';
import { NavLink } from 'react-router-dom';
import './SideBar.scss';

const SideBar = ({name, quote, rank}) => {
  return (
   <nav>
     <section className='user-info'>
       <image src='' alt='' />
       <h2>{name}</h2>
       <p>{quote}</p> 
       <p>{rank}</p>
     </section>

     <section className='nav-links'>
       <NavLink exact to='/movies' className='nav'> Movies </NavLink>
       <NavLink exact to='/characters' className='nav'> Characters </NavLink>
       <NavLink exact to='/favourites' className='nav'> Favourites </NavLink>
       <NavLink exact to='/' className='nav'> Sign out </NavLink>
     </section>
   </nav>
  )
}

export default SideBar;