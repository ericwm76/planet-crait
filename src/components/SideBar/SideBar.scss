$crait-color : #c0392b;
$background-color : #faebd745;

nav {
    width: 100%;
    margin: 20px;
    padding: 20px;
    background: $background-color;
    border: 2px solid $crait-color;
    border-radius: 10px;
    display: flex;
    justify-content: space-between;
}

.nav-links {
    display: flex;
    justify-content: space-evenly;
    align-items: center;
    width: 50%;
    .nav {
        text-decoration: none;
        color: white;
        font-weight: bold;
        border: 2px solid $crait-color;
        padding: 10px;
        border-radius: 10px;
        &:hover {
            background-color: #EAFCEB;
            cursor: pointer;
            box-shadow: 
                0 0 20px 5px #EAFCEB,  /* inner white */
                0 0 30px 10px #337F23, /* middle magenta */
                0 0 50px 20px #69A267; /* outer cyan */
            border: none;
            color: black;
          }
    }
}

.user-image {
  height: 5em;
  width: 5em;
  border-radius: 50%;
}

.user-info {
  display: flex;
}

.user-section {
  margin: 10px;
}