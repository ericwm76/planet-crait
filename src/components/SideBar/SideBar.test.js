import React from 'react';
import { shallow } from 'enzyme';
import SideBar from './SideBar';

describe('SideBar', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<SideBar />)
  })

  it('should match snapshot with all data passing in correctly', () => {
    expect(wrapper).toMatchSnapshot();
  })
})