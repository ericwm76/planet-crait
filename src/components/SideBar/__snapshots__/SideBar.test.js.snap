// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideBar should match snapshot with all data passing in correctly 1`] = `
<nav>
  <section
    className="user-info"
  >
    <div>
      <img
        alt="user default image"
        className="user-image"
        src="https://ksassets.timeincuk.net/wp/uploads/sites/55/2017/10/star-wars-8-porg-920x584.png"
      />
    </div>
    <div
      className="user-section"
    >
      <h2 />
      <p />
      <p />
    </div>
  </section>
  <section
    className="nav-links"
  >
    <NavLink
      className="nav"
      exact={true}
      to="/movies"
    >
       Movies 
    </NavLink>
    <NavLink
      className="nav"
      exact={true}
      to="/favourites"
    >
       Favourites 
    </NavLink>
    <NavLink
      className="nav"
      exact={true}
      to="/"
    >
       Sign out 
    </NavLink>
  </section>
</nav>
`;
